Podcast/Episode.swift

```diff
@@ -18,24 +19,79 @@ class Episode: NSObject {
     var descriptionText: String!
     var smallArtworkImageURL: URL?
     var largeArtworkImageURL: URL?
-    var smallArtworkImage: UIImage!
-    var largeArtworkImage: UIImage!
-    var mp3URL: URL?
-    var time: Double = 0
-    var tags: [Tag] = []
-    var nRecommended: Int = 0
-    var isBookmarked: Bool = false
-    var isRecommended: Bool = false
-    var isPlaying: Bool! = false
+    var smallArtworkImage: UIImage?
+    var largeArtworkImage: UIImage?
+    var mp3URL : URL?
+    var duration: Double = 0
+    var tags : [Tag] = []
+    var numberOfRecommendations = 0
+    var isBookmarked = false
+    var isRecommended = false
+    var isPlaying = false
     
+    init(id: Int) {
+        self.id = id 
+    }
+    
+    //all atribute initializer
+    init(id: Int, title: String = "", dateCreated: Date = Date(), descriptionText: String = "Not avaliable", smallArtworkImageURL: URL, largeArtworkImageURL: URL, mp3URL : String = "", duration: Double = 0, seriesTitle: String = "", tags: [Tag] = [], numberOfRecommendations: Int = 0, isRecommended: Bool = false, isBookmarked: Bool = false, isPlaying:Bool = false) {
+        self.id = id
+        self.title = title
+        self.dateCreated = dateCreated
+        self.descriptionText = descriptionText
+        self.smallArtworkImageURL = smallArtworkImageURL
+        self.largeArtworkImageURL = largeArtworkImageURL
+        self.mp3URL = URL(string: mp3URL)
+        self.isRecommended = isRecommended
+        self.isBookmarked = isBookmarked
+        self.isPlaying = isPlaying
+        self.numberOfRecommendations = numberOfRecommendations
+        self.seriesTitle = seriesTitle
+        self.duration = duration
+        self.tags = tags
 
-    init(id: Int, title: String = "", dateCreated: Date = Date(), descriptionText: String = "Not avaliable", smallArtworkImage: UIImage = #imageLiteral(resourceName: "filler_image"), largeArtworkImage: UIImage = #imageLiteral(resourceName: "filler_image"), mp3URL : String = "") {
+        super.init()
+    }
+    
+    //initializer without small/largeArtworkURL optionals
+    init(id: Int, title: String = "", dateCreated: Date = Date(), descriptionText: String = "Not avaliable", mp3URL : String = "", duration: Double = 0, seriesTitle: String = "", tags: [Tag] = [], numberOfRecommendations: Int = 0, isRecommended: Bool = false, isBookmarked: Bool = false, isPlaying:Bool = false) {
```

Comment:
why would you have an initializer without optionals? they are optional for a reason...